SofascoreAcademy/SofascoreAcademy/Views/MainViewController.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  MainViewController.swift
+//  SofascoreAcademy
+//
+//  Created by Akademija on 08.04.2024..
+//
+
+import Foundation
+import UIKit
+import SofaAcademic
+import SnapKit
+
+protocol parentSportSlugPicker {
+    func getPressedTab(request: sport_slug?)
+}
+
+protocol settingsButtonTap {
+    func reactToSetingsTap()
+}
+
+protocol tableCellTap {
+    func reactToCellTap(match: matchData)
+}
+
+enum sport_slug {
+    case football
+    case basketball
+    case american_football
+}
+
+class MainViewController : UIViewController {
+    
+    private let footballViewController = FootballViewController()
+    private let basketballViewController = BasketballViewController()
+    private let americanFootballViewController = AmericanFootballViewController()
+    
+    private let appHeader = AppHeader()
+    private let customTabBarController = CustomTabBarController()
+    
+    private let blueContainer = UIView()
+    private let containerView = UIView()
+    private var currentChild = UIViewController()
```

Comment:
```suggestion
    private var currentChild: UIViewController?
```

Mozes ovako, ne treba ti nova instanca nekog praznog UIViewController-a